src/compiler/utilities.ts

```diff
@@ -974,6 +974,13 @@ namespace ts {
 
     export function getErrorSpanForNode(sourceFile: SourceFile, node: Node): TextSpan {
         let errorNode: Node | undefined = node;
+
+        if (errorNode === undefined) {
+            // If we don't have a better node, then just set the error on the first token of
+            // construct.
+            return getSpanOfTokenAtPosition(sourceFile, 0);
+        }

```

Comment:
I don't think this is the appropriate fix. I think the right fix is to make sure that nobody is actually passing in `undefined`, so find the type assertions that are wrong.